python/lsst/pipe/supertask/cmdLineActivator.py

```diff
@@ -80,6 +81,39 @@ def _printTable(rows, header):
     for col1, col2 in rows:
         print(col1.ljust(width), col2)
 
+
+class _MPMap(object):
+    """Class implementing "map" function using multiprocessing pool.
+
+    Parameters
+    ----------
+    numProc : `int`
+        Number of process to use for executing tasks.
+    timeout : `float`
+        Time in seconds to wait for tasks to finish.
+    """
+
+    def __init__(self, numProc, timeout):
+        self.numProc = numProc
+        self.timeout = timeout
+
+    def __call__(self, function, iterable):
+        """Apply function to every item of iterable.
+
+        Wrapper around pool.map_async, to handle timeout. This is required
+        so as to trigger an immediate interrupt on the KeyboardInterrupt
+        (Ctrl-C); see
+        http://stackoverflow.com/questions/1408356/keyboard-interrupts-with-pythons-multiprocessing-pool
+
+        Further wraps the function in _poolFunctionWrapper to catch exceptions
+        that don't inherit from Exception.
+        """
+        disableImplicitThreading()  # To prevent thread contention
+        import multiprocessing
```

Comment:
That was copied from CmdLineTask. I thought more about it and I think it's better indeed to move it to the top of the module.